wave_matching/include/wave/matching/matcher.hpp

```diff
@@ -0,0 +1,51 @@
+#ifndef WAVE_MATCHING_MATCHER_HPP
+#define WAVE_MATCHING_MATCHER_HPP
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+namespace wave {
+
+// sensor data type
+template <typename T>
+class Matcher {
+ public:
```

Comment:
commenting here so it's tracked:

We've used 1 for autonomoose to keep cpplint from spewing piles except where we truly do deviate.

Depends solely on your desired level of cpplint adherence.